portal-impl/src/com/liferay/portal/service/impl/CountryServiceImpl.java

```diff
@@ -19,20 +19,42 @@
 import com.liferay.portal.kernel.exception.CountryIddException;
 import com.liferay.portal.kernel.exception.CountryNameException;
 import com.liferay.portal.kernel.exception.CountryNumberException;
+import com.liferay.portal.kernel.exception.NoSuchCountryException;
 import com.liferay.portal.kernel.exception.PortalException;
 import com.liferay.portal.kernel.model.Country;
 import com.liferay.portal.kernel.security.access.control.AccessControlled;
 import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.util.OrderByComparator;
 import com.liferay.portal.kernel.util.Validator;
 import com.liferay.portal.service.base.CountryServiceBaseImpl;
 
 import java.util.List;
+import java.util.Map;
 
 /**
  * @author Brian Wing Shun Chan
  */
 public class CountryServiceImpl extends CountryServiceBaseImpl {
 
+	@Override
+	public Country addCountry(
+			boolean active, String a2, String a3, boolean billingAllowed,
```

Comment:
See my comments in the local service, the same applies to the `add` and `update` methods here.